src/main_benchmark.cpp

```diff
@@ -0,0 +1,411 @@
+    std::transform(dimsString.begin(), dimsString.end(), std::back_inserter(shape),
+                                   [](const std::string& s) -> std::size_t {
+        auto dimOpt = ovms::stoi64(s);
+        if (!dimOpt.has_value()) {
+            std::cout << __LINE__ << " " << s << std::endl;
+            throw std::invalid_argument("Invalid shape argument");
+        }
+                                   return dimOpt.value(); });
+    return shape;
+}
+
+volatile sig_atomic_t shutdown_request = 0;
+
+static void onInterrupt(int status) {
+    shutdown_request = 1;
+}
+
+static void onTerminate(int status) {
+    shutdown_request = 1;
+}
+
+static void onIllegal(int status) {
+    shutdown_request = 2;
+}
+
+static void installSignalHandlers() {
+    static struct sigaction sigIntHandler;
+    sigIntHandler.sa_handler = onInterrupt;
+    sigemptyset(&sigIntHandler.sa_mask);
+    sigIntHandler.sa_flags = 0;
+    sigaction(SIGINT, &sigIntHandler, NULL);
+
+    static struct sigaction sigTermHandler;
+    sigTermHandler.sa_handler = onTerminate;
+    sigemptyset(&sigTermHandler.sa_mask);
+    sigTermHandler.sa_flags = 0;
+    sigaction(SIGTERM, &sigTermHandler, NULL);
+
+    static struct sigaction sigIllHandler;
+    sigIllHandler.sa_handler = onIllegal;
+    sigemptyset(&sigIllHandler.sa_mask);
+    sigIllHandler.sa_flags = 0;
+    sigaction(SIGILL, &sigIllHandler, NULL);
+}
+
+using shape_t = std::vector<size_t>;
+
+OVMS_InferenceRequest* prepareRequest(const std::string& modelName, uint64_t modelVersion, OVMS_DataType datatype, const shape_t& shape, const std::string& inputName, const void* data) {
+    OVMS_InferenceRequest* request{nullptr};
+    OVMS_InferenceRequestNew(&request, modelName.c_str(), modelVersion);
+    OVMS_InferenceRequestAddInput(request, inputName.c_str(), datatype, shape.data(), shape.size());
+    size_t elementsCount = std::accumulate(shape.begin(), shape.end(), 1, std::multiplies<size_t>());
+    OVMS_InferenceRequestInputSetData(request, inputName.c_str(), data, sizeof(float) * elementsCount, OVMS_BUFFERTYPE_CPU, 0);  // TODO sizeof
+    return request;
+}
+
+void triggerInferenceInALoop(
+    std::future<void>& startSignal,
+    std::future<void>& stopSignal,
+    const size_t niterPerThread,
+    size_t& wholeThreadTimeUs,
+    double& averageWholeLatency,
+    double& averagePureLatency,
+    OVMS_Server* server,
+    OVMS_InferenceRequest* requests) {  // TODO each thread should get 2 to switch
+    OVMS_InferenceResponse* response{nullptr};
+    startSignal.get();
+    std::vector<uint64_t> latenciesWhole(niterPerThread);
+    std::vector<uint64_t> latenciesPure(niterPerThread);
+    auto workloadStart = std::chrono::high_resolution_clock::now();
+    size_t iter = niterPerThread;
+    while (iter-- > 0) {
+        auto iterationStart = std::chrono::high_resolution_clock::now();
+        stopSignal.wait_for(std::chrono::milliseconds(0));
+        // TODO add here resetting buffer
```

Comment:
Why not now?

Reply:
In essence we are recreating OV::Tensor anyway each time that we call Inference(). However this could be part of 3 modes support:
1) Full inference request creation path
2) Just resetting buffers
3) No resetting buffers